Sources/MapboxCoreNavigation/SDKVersionProvider.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+struct VersionProvider {
+    enum MapboxSDK {
+        case mapboxNavigation
+        case mapboxCoreNavigation
+    }
+
+    private static let bundleShortVersionKey = "CFBundleShortVersionString"
+
+    private static var cache = [MapboxSDK: String]()
+
+    static func version(for SDK: MapboxSDK) -> String {
+        if let sdkVersion = cache[SDK] {
+            return sdkVersion
+        }
+
+        let sdkVersion = resolveVersion(for: SDK)
+        cache[SDK] = sdkVersion
+
+        return sdkVersion
+    }
+
+    private init() {}
+
+    private static func resolveVersion(for SDK: MapboxSDK) -> String {
+        // Checks if MBXInfo.plist exists
+        if let infoDictionary = infoDictionary(for: SDK),
+           let sdkVersion = infoDictionary[bundleShortVersionKey] as? String {
+            return sdkVersion
+        }
+
+        // If MBXInfo.plist wasn't included, try to parse version from SDK bundle's Info.plist
+        guard let SDKBundle = bundle(for: SDK) else {
+            preconditionFailure("Navigation SDK bundle cannot be resvoled")
```

Comment:
Technically should never happen (probably except for very rare tricks with `dlopen` which no one will use).